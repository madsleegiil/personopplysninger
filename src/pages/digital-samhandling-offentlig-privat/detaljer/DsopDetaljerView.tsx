import React from "react";
import { Normaltekst, Undertittel } from "nav-frontend-typografi";
import ListElement from "components/listelement/ListElement";
import { FormattedMessage } from "react-intl";
import FileSaver from "file-saver";
import moment from "moment";
import { Hovedknapp } from "nav-frontend-knapper";
import { DsopInfo } from "../../../types/dsop";

interface Props {
  id: string;
  dsopInfo: DsopInfo;
}

const DsopDetaljerView = (props: Props) => {
  const { id, dsopInfo } = props;
  const innslag = dsopInfo.filter((d) => d.uthentingsTidspunkt === id).shift();

  return innslag ? (
    (() => {
      const onClick = () => {
        const leverteData = JSON.parse(atob(innslag.leverteData));
        const fileContent = JSON.stringify(leverteData, null, 2);
        const fileBlob = new Blob([fileContent], {
          type: "application/json",
        });
        const fileName = "utleverte-data.json";
        FileSaver.saveAs(fileBlob, fileName);
      };
      return (
        <div>
          <div className="detaljer__tittel">
            <Undertittel>{innslag.mottakernavn}</Undertittel>
            <Normaltekst>
              <FormattedMessage
                id="side.organisasjonsnummer"
                values={{ orgnr: innslag.mottaker }}
              />
            </Normaltekst>
          </div>
          <hr className="box__linje-bred" />
          <div className="box">
            <div className="box__content">
              <ul className="list-column-2">
                <ListElement
                  titleId={"dsop.uthentingstidspunkt"}
                  content={moment(innslag.uthentingsTidspunkt).format(
                    "DD.MM.YYYY hh:mm"
                  )}
                />
                <ListElement
                  titleId={"dsop.personnummer"}
                  content={innslag.person}
                />
                <ListElement titleId={"dsop.tema"} content={innslag.tema} />
                <ListElement
                  titleId={"dsop.behanglingsGrunnlag"}
                  content={innslag.behandlingsGrunnlag}
                />
              </ul>
            </div>
            <div className="detaljer__container">
              <Hovedknapp onClick={onClick}>
                <FormattedMessage id="dsop.lastned" />
              </Hovedknapp>
            </div>
          </div>
        </div>
      );
    })()
  ) : (
    <div>
      <FormattedMessage id="dsop.ingendata" />
    </div>
  );
};

export default DsopDetaljerView;
