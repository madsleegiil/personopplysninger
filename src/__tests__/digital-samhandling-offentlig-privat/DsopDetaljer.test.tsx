import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import DsopDetaljerView from "../../pages/digital-samhandling-offentlig-privat/detaljer/DsopDetaljerView";
import dsopInfo from "../../clients/apiMock/app/fetch/dsop-info.json";
import { IntlProvider } from "react-intl";
import nbMessages from "text/nb";

describe("DsopDetaljerView", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <IntlProvider locale={"nb"} messages={nbMessages}>
          <DsopDetaljerView
            dsopInfo={dsopInfo}
            id={"2019-06-18T10:45:51.634"}
          />
        </IntlProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
