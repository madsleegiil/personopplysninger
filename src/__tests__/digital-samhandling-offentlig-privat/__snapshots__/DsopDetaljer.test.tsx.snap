// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DsopDetaljerView renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="detaljer__tittel"
    >
      <h2
        class="navds-heading navds-heading--small"
      >
        STOREBRAND LIVSFORSIKRING AS
      </h2>
      <p
        class="navds-body-short"
      >
        Organisasjonsnummer 958995369
      </p>
    </div>
    <hr
      class="box__linje-bred"
    />
    <div
      class="box"
    >
      <div
        class="box__content"
      >
        <ul
          class="list-column-2"
        >
          <li>
            <div
              class="listelement__header"
            >
              <p
                class="navds-label"
              >
                Uthentingstidspunkt
              </p>
            </div>
            <div
              class="content "
            >
              18.06.2019 10:45
            </div>
            <p
              class="navds-body-short"
            />
          </li>
          <li>
            <div
              class="listelement__header"
            >
              <p
                class="navds-label"
              >
                Personnummer
              </p>
            </div>
            <div
              class="content "
            >
              04019727454
            </div>
            <p
              class="navds-body-short"
            />
          </li>
          <li>
            <div
              class="listelement__header"
            >
              <p
                class="navds-label"
              >
                Tema
              </p>
            </div>
            <div
              class="content "
            >
              AAP
            </div>
            <p
              class="navds-body-short"
            />
          </li>
          <li>
            <div
              class="listelement__header"
            >
              <p
                class="navds-label"
              >
                Behandlingsgrunnlag
              </p>
            </div>
            <div
              class="content "
            >
              Samtykke
            </div>
            <p
              class="navds-body-short"
            />
          </li>
        </ul>
      </div>
      <div
        class="detaljer__container"
      >
        <button
          class="navds-button navds-button--primary navds-button--medium"
        >
          <span
            aria-live="polite"
            class="navds-label"
          >
            Last ned leverte data
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
